src/jpgimage.cpp

```diff
@@ -73,10 +75,15 @@ constexpr Internal::TagDetails jpegProcessMarkerTags[] = {
     {sof15_, N_("Lossless, differential arithmetic coding")},
 };
 
-constexpr auto exifId_ = "Exif\0\0";  //!< Exif identifier
-// constexpr auto jfifId_ = "JFIF\0";                         //!< JFIF identifier
-constexpr auto xmpId_ = "http://ns.adobe.com/xap/1.0/\0";  //!< XMP packet identifier
-constexpr auto iccId_ = "ICC_PROFILE\0";                   //!< ICC profile identifier
+constexpr std::array<byte, 6> exifId_{
+    0x45, 0x78, 0x69, 0x66, 0x00, 0x00,
+};  //!< Exif identifier
+constexpr std::array<byte, 29> xmpId_{
+    0x68, 0x74, 0x74, 0x70, 0x3A, 0x2F, 0x2F, 0x6E, 0x73, 0x2E, 0x61, 0x64, 0x6F, 0x62, 0x65,
+    0x2E, 0x63, 0x6F, 0x6D, 0x2F, 0x78, 0x61, 0x70, 0x2F, 0x31, 0x2E, 0x30, 0x2F, 0x00,
+};  //!< XMP packet identifier
```

Comment:
I don't like this change. These were readable strings before, but now they look like binary data.

Reply:
How about switching to '' notation?